src/Views/OperatingSystemView.vala

```diff
@@ -248,6 +257,33 @@ public class About.OperatingSystemView : Gtk.Grid {
         }
     }
 
+    private bool confirm_reboot_to_firmware_setup_action () {
+        var dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Are you sure you want to Reboot?"),
+            _("This will close all open applications and turn off this device."),
```

Comment:
We can probably be more specific here
```suggestion
            _("Restart to firmware setup?"),
            _("This will close all open applications, restart this device, and open the firmware setup screen."),
```